freqtrade/strategy/hyper.py

```diff
@@ -244,8 +244,8 @@ def enumerate_parameters(self, category: str = None) -> Iterator[Tuple[str, Base
             if not attr_name.startswith('__'):  # Ignore internals, not strictly necessary.
                 attr = getattr(self, attr_name)
                 if issubclass(attr.__class__, BaseParameter):
-                    if category is None or category == attr.category or \
-                       attr_name.startswith(category + '_'):
+                    if (category is None or category == attr.category or
+                            (attr_name.startswith(category + '_') and attr.category is None)):
```

Comment:
I think this should be an error. People will do a stupid thing and not realize why their parameter is silently excluded from optimization.

Reply:
i would almost tend to leave it this way for now and eventually implement this at a later point IF we see it's a probem.

i don't expect people to actually use both space and naming convention, but mainly rely on naming convention ...

Reply:
Right, we do not expect them, but they eventually will. You are case and point :]

Reply:
It's failing now in this case ;)